### PR TITLE
Rename `Mesh` to `MeshXY`

### LINKED ISSUE #6052: Refactor `Mesh` to `MeshXY`
Prepares us for https://github.com/SciTools/iris/discussions/5994 later on - avoiding that being a breaking change.

### PR DESCRIPTION
**Pull request by trexfeathers**
### Description

Closes #6052.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

---
Add any of the below labels to trigger actions on this PR:

- https://github.com/SciTools/iris/labels/benchmark_this

### CODE REVIEW

**Comment by trexfeathers** [comment]
I have pushed up the rename commit on its own to hopefully demonstrate that everything passes 🤞

I will then push up another commit that adds a placeholder `Mesh` class and makes `MeshXY` a subclass of that.

File: lib/iris/experimental/ugrid/mesh.py [hunk 1]
```diff
@@ -579,20 +579,60 @@ def xml_element(self, doc):
...
+    @abstractmethod
+    def __init__(self):
+        message = (
+            f"The {self.__class__.__name__} class is not yet implemented. "
+            "Use the MeshXY class instead."
+        )
+        raise NotImplementedError(message)
+
+
+class MeshXY(Mesh):
```
Comment (trexfeathers) [inline lib/iris/experimental/ugrid/mesh.py:617]: Note that there are still some string references to the word "Mesh" throughout the code base. I changed those that were making specific reference to the class, but I did not change those that were referring conceptually to a UGRID Mesh. Hopefully this leaves our documentation in a better place for the future ambition of offering multiple types of `Mesh` class.

File: docs/src/further_topics/ugrid/operations.rst [hunk 2]
```diff
@@ -60,8 +60,26 @@ subsequent example operations on this page.
...
+
+                >>> from iris.coords import AuxCoord
+                >>> from iris.experimental.ugrid import Connectivity, MeshXY
+
+                # Going to create the following mesh
+                #  (node indices are shown to aid understanding):
+                #
+                #  0----1
+                #  |    |
+
```
Comment (HGWright) [inline docs/src/further_topics/ugrid/operations.rst:80]: Hey @trexfeathers, this appears to be here by mistake. Could you remove it please?

**Review by HGWright — CHANGES_REQUESTED** [review]

File: docs/src/further_topics/ugrid/operations.rst (hunk 2 above)
Comment (trexfeathers) [reply docs/src/further_topics/ugrid/operations.rst]: ea7821743d6fe8ed4cbccc729cdadc1ecea68880

**Review by HGWright — APPROVED** [review]
Thanks @trexfeathers for this, definitely makes this clearer going forward.

**Comment by trexfeathers** [comment]
Thanks @HGWright!

**Pull request merged**